src/test/java/de/tum/in/www1/artemis/quiz/QuizExerciseIntegrationTest.java

```diff
@@ -1837,6 +2004,61 @@ void testReset() throws Exception {
         }
     }
 
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void createQuizExercise_dragAndDrop_withoutBackgroundFile() throws Exception {
+        Course course = database.createCourse();
+        QuizExercise quizExercise = database.createQuiz(course, ZonedDateTime.now().plusHours(5), null, QuizMode.SYNCHRONIZED);
+        quizExercise.setDuration(3600);
+        createQuizExerciseWithFiles(quizExercise, HttpStatus.CREATED, false);
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void createQuizExercise_withoutDragAndDrop() throws Exception {
+        Course course = database.createCourse();
+        QuizExercise quizExercise = database.createQuiz(course, ZonedDateTime.now().plusHours(5), null, QuizMode.SYNCHRONIZED);
+        quizExercise.setQuizQuestions(quizExercise.getQuizQuestions().stream().filter(question -> !(question instanceof DragAndDropQuestion)).toList());
+        quizExercise.setDuration(3600);
+        createQuizExerciseWithFiles(quizExercise, HttpStatus.CREATED, false);
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void updateQuizExercise_withoutDragAndDrop() throws Exception {
+        Course course = database.createCourse();
+        QuizExercise quizExercise = database.createQuiz(course, ZonedDateTime.now().plusHours(5), null, QuizMode.SYNCHRONIZED);
+        quizExercise.setQuizQuestions(quizExercise.getQuizQuestions().stream().filter(question -> !(question instanceof DragAndDropQuestion)).toList());
+        quizExercise.setDuration(3600);
+        quizExercise = createQuizExerciseWithFiles(quizExercise, HttpStatus.CREATED, false);
+        updateQuizExerciseWithFiles(quizExercise, null, HttpStatus.OK);
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void updateQuizExercise_dragAndDrop_withoutFileArrayProvided() throws Exception {
+        QuizExercise quizExercise = createQuizOnServer(ZonedDateTime.now().plusHours(5), null, QuizMode.SYNCHRONIZED);
```

Comment:
do you need to use createQuizOnServer? I think the method is a bit slow, I can recommend `database.createAndSaveQuiz(...)` instead (just saves it in the repo without any POST requests).

Reply:
The method doesn't handle saving the images to the server right now, so the tests produce errors.
It's not quite within the scope of my PR to adapt the Utils method (it's not that easy atm).

Therefore, I'd like to stick with this approach for now.